src/modules/newtek/util/ndi.cpp

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2018
+ *
+ * This file is part of CasparCG (www.casparcg.com).
+ *
+ * CasparCG is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * CasparCG is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with CasparCG. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Author: Krzysztof Zegzula, zegzulakrzysztof@gmail.com
+ */
+
+#include "../StdAfx.h"
+
+#include "ndi.h"
+
+#include <memory>
+#include <mutex>
+
+#ifdef _WIN32
+#include <windows.h>
+#else
+#include <dlfcn.h>
+#include <stdlib.h>
+#endif
+
+#include <common/except.h>
+
+#include <boost/filesystem.hpp>
+
+namespace caspar { namespace newtek { namespace ndi {
+
+const std::wstring& dll_name()
+{
+    static std::wstring name = u16(NDILIB_LIBRARY_NAME);
+
+    return name;
+}
+
+static std::mutex                              find_instance_mutex;
+static std::shared_ptr<NDIlib_find_instance_t> find_instance;
+
+NDIlib_v3* load_library()
+{
+    static NDIlib_v3* ndi_lib = nullptr;
+
+    if (ndi_lib)
+        return ndi_lib;
+    const char* runtime_dir = getenv(NDILIB_REDIST_FOLDER);
+
+    if (runtime_dir == NULL)
+        return nullptr;
```

Comment:
AFAIK the DLLs can be legally obtained only by installing NDI Redistributable (or other NewTek's tools), and they are backward compatible so multiple apps can use different API versions off the latest DLL. I think we should stick to it, but if you wish, I can make it as you suggest.

Reply:
That argument is mostly aimed at Linux as there is no installer provided, you get a zip with some files in. So the easiest option there is to dump them in the folder with other libs rather than set it up as a system wide lib.

Also, by using the local dir we also allow users to make easier to deploy versions of caspar. I can see this being especially valuable when deploying to large numbers of servers, as then the libs are versioned with caspar.

So yes, I would like it try the local dir first, then try the system default location.

Reply:
Done - now the local directory is checked first. (Not tested on Linux yet)